webots_ros2_driver/CMakeLists.txt

```diff
@@ -30,62 +30,75 @@ find_package(vision_msgs REQUIRED)
 find_package(webots_ros2_msgs REQUIRED)
 find_package(tinyxml2_vendor REQUIRED)
 find_package(TinyXML2 REQUIRED)
-find_package(yaml-cpp REQUIRED)
+find_package(yaml_cpp_vendor REQUIRED)
 
-if($ENV{ROS_DISTRO} MATCHES "humble")
-  find_package(Python 3.10 EXACT REQUIRED COMPONENTS Development)
-elseif($ENV{ROS_DISTRO} MATCHES "iron")
-  find_package(Python 3.10 EXACT REQUIRED COMPONENTS Development)
-elseif($ENV{ROS_DISTRO} MATCHES "jazzy")
-  find_package(Python 3.12 EXACT REQUIRED COMPONENTS Development)
-elseif($ENV{ROS_DISTRO} MATCHES "kilted")
-  find_package(Python 3.12 EXACT REQUIRED COMPONENTS Development)
-elseif($ENV{ROS_DISTRO} MATCHES "rolling")
-  find_package(Python 3.12 EXACT REQUIRED COMPONENTS Development)
+if($ENV{ROS_DISTRO} MATCHES "humble|iron")
+    set(PYTHON_VER 3.10)
+else()
+    set(PYTHON_VER 3.12)
 endif()
+find_package(Python ${PYTHON_VER} EXACT REQUIRED COMPONENTS Development)
 
-add_custom_target(compile-lib-controller ALL
-  COMMAND ${CMAKE_COMMAND} -E env "WEBOTS_HOME=${CMAKE_CURRENT_SOURCE_DIR}/webots" make release -f Makefile > /dev/null 2>&1
-  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/webots/src/controller
-)
-
-add_custom_target(compile-lib-vehicle ALL
-  COMMAND ${CMAKE_COMMAND} -E env "WEBOTS_HOME=${CMAKE_CURRENT_SOURCE_DIR}/webots" make release -f Makefile > /dev/null 2>&1
-  DEPENDS compile-lib-controller
-  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/webots/projects/default/libraries/vehicle
-)
 
-add_custom_target(compile-generic-window ALL
-  COMMAND ${CMAKE_COMMAND} -E env "WEBOTS_HOME=${CMAKE_CURRENT_SOURCE_DIR}/webots" make release -f Makefile > /dev/null 2>&1
-  DEPENDS compile-lib-controller
-  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/webots/resources/projects/libraries/generic_robot_window
-)
+# <<< CROSS-PLATFORM CHANGE: detect prebuilt Webots.app on macOS
```

Comment:
This "Change"-comments does not make a lot of sense in long lived files, probably this can be removed?